evap/staff/importers/json.py

```diff
@@ -546,22 +578,21 @@ def _import_contribution(self, evaluation: Evaluation, data: ImportRelated) -> t
         return contribution, created
 
     def _import_events(self, data: list[ImportEvent]) -> None:  # noqa:PLR0912
-        # Divide in two lists so corresponding courses are imported before their exams
-        non_exam_events = (event for event in data if not event["isexam"])
-        exam_events = (event for event in data if event["isexam"])
+        # Divide in multiple lists to handle individually
+        non_exam_events, exam_events, exam_events_without_related_non_exam_event = [], [], []
+        for event in data:
+            if not event["isexam"]:
+                non_exam_events.append(event)
+            elif event.get("relatedevents"):
+                exam_events.append(event)
+            else:
+                exam_events_without_related_non_exam_event.append(event)
 
+        # Only import courses in first step, don't create evaluations yet
         for event in non_exam_events:
             course = self._import_course(event)
```

Comment:
We are not using the course in this loop anymore
```suggestion
            self._import_course(event)
```